Contributors config and design

ymschaap opened the pull request:
Re: #226
Re: #214

To speed things up, I've taken the spreadsheet here:
> https://docs.google.com/spreadsheets/d/1LR29JpI2XJSjan_z6uP1YpRy4EYYivYZPqCIG-9sfFY/edit?ts=5d13c7d0#gid=770177464

And transformed it to json using a script.

If we add the bio's from #239 as a column I could quickly update this file to also include those. Else we have to manually copy+pasta.

I did lose any existing edits, and hence the fields: tagline, gravatar, twitter, etc. We could manually add that (I think there were ~6 contributors).

Given we have github accounts for 66 of the users, I can run it through the Github User API and extract that image & bio. Ok?

**Update:** I did that just now, let me know if that works (e.g. maybe contributors don't expect the Github bio to appear here).

rviscomi commented:
Cool! Thanks for working on this @ymschaap!

A few suggestions:

- use the contributor's GitHub ID as the default ID (fall back to first/last name if no GitHub ID is given)
- remove `bio` field since that only applies to authors
- for any existing contributors in the config, try to merge their previous info like twitter fields and taglines

bazzadp commented:
> use the contributor's GitHub ID as the default ID (fall back to first/last name if no GitHub ID is given)

Any reason why? I think it's actually nicer being firstname-lastname like @ymschaap has done. Better for searchability and SEO too. But then I'm probably weird cause I don't really use my GitHub handle (@bazzadp) in my other online profiles (I really should change it but don't want to lose my history). On plus die it would allow consistent naming convention even for those not on GitHub. On the downside it would mean editing all the chapters to alter the first section but don't think that's too difficult.

rviscomi commented:
No particular reason. I can be convinced. But yes it would require syncing with the chapter metadata.

ymschaap commented:
@rviscomi 
* implemented changes (by github Id as key, if available, bio gone, merge previous added missing contributors (1))
* I don't have access to the invisionapp mockup document, so I just winged it.
* I tried to sort all contributors by name (instead of key), maybe anybody could help me sort by name key here ?
* I did some UX work on the button functionality
* I dropped tagline and twitter (we don't have twitter handles, and the tagline wasn't in the contributor.html yet)
*

rviscomi commented:
Thanks for working on this.

Shared the Invision designs with you.

I'd like to keep the `twitter` field. The old config had it in a few places and could be merged with the new GitHub-based data.

rviscomi commented on src/templates/en/2019/contributors.html:
```diff
@@ -107,20 +171,21 @@ <h1>
     <div class="contributor-grid">
     {% for id, contributor in config.contributors.items() %}
       <div class="contributor {{ ' '.join(contributor.teams) }}">
-        <img class="contributor-avatar" src="{{ contributor.avatar_url }}" height="100" width="100" alt="{{ contributor.name }} avatar">
```
`avatar_url` is a pre-generated URL that falls back to a random character illustration from the Almanac design. We should try to keep that functionality.

rviscomi commented on src/templates/en/2019/contributors.html:
```diff
@@ -26,31 +26,63 @@
 .contributor-grid {
   display: flex;
   align-items: stretch;
+  flex-direction: row;
```
On mobile, contributors should appear one per row.

![image](https://user-images.githubusercontent.com/1120896/67813970-3fce9200-fa79-11e9-9a53-ea10fa655814.png)

rviscomi commented on src/templates/en/2019/contributors.html:
```diff
@@ -107,20 +171,21 @@ <h1>
+{% endif %}" height="100" width="100" alt="{{ contributor.name }} avatar" loading="lazy">
         <div class="contributor-name">{{ contributor.name }}</div>
         {% if contributor.twitter or contributor.github %}
         <div class="contributor-social">
```
Some contributors have no social links. If possible this space would be empty so the team names are aligned across the row of contributors.

![image](https://user-images.githubusercontent.com/1120896/67814053-7c01f280-fa79-11e9-8d69-4905e024319f.png)

ymschaap commented:
Implemented your feedback.
